microsoft/vscode-azureappservice
# Create workspace based app insights

Leverages log analytics create step, which is required to migrate App Insights to the Workspace Based. Here is some [documentation](https://learn.microsoft.com/en-us/azure/azure-monitor/app/convert-classic-resource#migrate-your-resource) on that. This was already being done in Functions, so the step already existed.

It also uses the connection string property rather than the instrumentational key since that is the recommended path going forward.

Another reference: https://learn.microsoft.com/en-us/azure/azure-monitor/app/create-workspace-resource

How to test:
1. Right-click "App Service" node and select "Create Web App..."
2. Fill out the prompts
3. Check to see if a log analytics workspace is created and linked to app insights
4. Check in the portal for the app insight component and make sure it's workspace-based (it'll tell you that you need to migrate it)
5. Try it with advanced too
